### PR TITLE
Aws sso

### PR DESCRIPTION
**Pull request by dividehex**
AWS SAML SSO rule

### CODE REVIEW

File: rules/awsSaml.js [hunk 1]
```diff
@@ -0,0 +1,261 @@
+function awsSaml(user, context, callback) {
+  // Only exec this rule if logging into AWS Saml SSO
+  if (context.clientID !== "q8vmyUjlPZj7QnURDVXtjvzSlsXK9AHK")
```
Comment (dividehex) [inline rules/awsSaml.js:3]: Before this is deployed to Prod, I'll need to add the Prod clientID.  Currently, this is the Dev clientID

File: rules/awsSaml.js [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+  function fetchUsersAWSGroups(userUUID) {
+    var params = {
+      IdentityStoreId: IdentityStoreId,
+      MemberId: {
+        UserId: userUUID,
+      },
+      MaxResults: 50,
+    };
+    // TODO: handle pagenation!!!
```
Comment (dividehex) [inline rules/awsSaml.js:144]: We currently don't have any users with over 50 AWS associated groups so not iterating past 50 should not be a problem at this time, but I will definitely need to properly handle paginated returns before we call this project complete.

File: rules/awsSaml.js [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    "team_netops",
+    "team_opsec",
+    "team_se",
+    "team_secops",
+    "voice-dev",
+    "vpn_sumo_aws_devs",
+  ];
+
+  // Filter the users Auth0 groups down to only those mapped to AWS groups
+  function filterAWSGRoups(groups) {
```
Comment (floatingatoll) [inline rules/awsSaml.js:82]: Minor capitalization nitpick for "Groups" here.

**Review by floatingatoll — APPROVED** [review]
Reviewed this weekend. Reasoning is sound and the code flow makes sense.

**Review by floatingatoll — APPROVED** [review]

**Review by floatingatoll — APPROVED** [review]

**Review by floatingatoll — APPROVED** [review]
Reasonable changes. Every iam site is in a different region.

**Pull request merged**